### ISSUE TITLE
[ENHANCEMENT] rke2 spegel support

### ISSUE THREAD

**Issue by ibrokethecloud** [harvester/harvester#6188]
**Is your enhancement related to a problem? Please describe.**

rke2 v1.27.x packages spegel registry as an addon feature.

https://docs.rke2.io/reference/server_config#experimental
https://github.com/spegel-org/spegel/

Harvester should investigate using the same for image loading as this would make the install / upgrade process quicker.

The iso would only be required to load the iso's on the first node and subsequent nodes could pull images from the first node.

As part of the investigation we should cover the following:
* changes needed in installer to enable spegel
* HA for spegel to handle loss of first node
* changes in installer for loading images during install and upgrade path

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by innobead** [comment]
@starbops should this just supersede https://github.com/harvester/harvester/pull/6181? 

cc @bk201

**Comment by innobead** [comment]
Comment from @ibrokethecloud, is reasonable for me. 
> it is experimental support in rke2 and we need time to figure out ha etc for it. suggest keeping what we are working on.

**Comment by starbops** [comment]
From the upgrade's perspective, assuming that Spegel is enabled, we could add it as a new option to the newly introduced `image-preload-strategy` setting, say, `spegel`. BTS, it will only preload images on the first node. Other nodes could then get the images on demand from the first node. Preloading images is still essential when it comes to air-gapped setups (if no private registries are involved).

**Comment by ibrokethecloud** [comment]
yeah pre-loading would be needed but only on the first node or first few nodes, rest of cluster could pull images from these nodes and this would speed up install / upgrade when you plan to have large clusters

**Comment by w13915984028** [comment]
The benchmark https://github.com/spegel-org/spegel/blob/main/docs/BENCHMARK.md#analysis looks great. With this embedded, cluster installation and upgrade can all benefit.

**Comment by phillebaba** [comment]
If you are preloading the images on a single node you probably want to make sure to pin them so that they are not garbage collected. I have been thinking about different solution to make this simple, like adding push functionality and having Spegel manage image replication among themselves. That is however further in the future.

I am not sure how RKE2 or K3S sees things but Spegel is currently pretty stable. Having Spegel  run directly on the host like RKE2 does it is probably the best solution, but it depends on the project committing to continue importing Spegel.

**Comment by starbops** [comment]
Hi @phillebaba, thank you for the reminder. Regarding the image pinning, we might have to wait until we bump the internal RKE2 to v1.29.x due to a bug fix available after Kubernetes v1.29.0 (kubernetes/kubernetes#119986)

**Comment by zackbradys** [comment]
Bumping this RFE! It would be great to start by having it as an optional experimental feature so users can enable while it is being investigated for its larger place in Harvester.
